Release ethosu driver after use.

hsharma35 opened the pull request:
Summary: Release ethosu driver in ArmBackendEthosU after execution. This will allow the backend to be used more than once.

Differential Revision: D61733840

zonglinpengmeta commented on backends/arm/runtime/ArmBackendEthosU.cpp:
```diff
@@ -164,8 +165,10 @@ class ArmBackend final : public PyTorchBackendInterface {
+    auto driver =
+        std::unique_ptr<ethosu_driver, decltype(&ethosu_release_driver)>(
+            ethosu_reserve_driver(), ethosu_release_driver);
+    if (!driver) {
```
Why not keep the check against null explicit :)

hsharma35 committed:
Release ethosu driver after use. (#4905)

Summary:
Pull Request resolved: https://github.com/pytorch/executorch/pull/4905

Release ethosu driver in ArmBackendEthosU after execution. This will allow the backend to be used more than once.

Reviewed By: digantdesai, zonglinpengmeta

Differential Revision: D61733840
